cli/src/config.rs

```diff
@@ -229,39 +227,54 @@ struct UnresolvedConfigEnv {
 }
 
 impl UnresolvedConfigEnv {
-    fn resolve(self) -> Result<ConfigPath, ConfigEnvError> {
-        if let Some(path) = self.jj_config {
-            // TODO: We should probably support colon-separated (std::env::split_paths)
-            return Ok(ConfigPath::new(Some(PathBuf::from(path))));
+    fn resolve(self) -> Result<Vec<ConfigPath>, ConfigEnvError> {
+        if let Some(paths) = self.jj_config {
+            return Ok(split_paths(&paths)
+                .map(|path| ConfigPath::new(Some(path)))
+                .collect());
         }
-        // TODO: Should we drop the final `/config.toml` and read all files in the
-        // directory?
-        let platform_config_path = ConfigPath::new(self.config_dir.map(|mut config_dir| {
-            config_dir.push("jj");
-            config_dir.push("config.toml");
-            config_dir
-        }));
+        let mut paths = vec![];
         let home_config_path = ConfigPath::new(self.home_dir.map(|mut home_dir| {
             home_dir.push(".jjconfig.toml");
             home_dir
         }));
+        let platform_config_path =
+            ConfigPath::new(self.config_dir.clone().map(|mut config_dir| {
+                config_dir.push("jj");
+                config_dir.push("config.toml");
+                config_dir
+            }));
+        let platform_config_dir = ConfigPath::new(self.config_dir.map(|mut config_dir| {
+            config_dir.push("jj");
+            config_dir.push("conf.d");
+            config_dir
+        }));
         use ConfigPath::*;
-        match (platform_config_path, home_config_path) {
-            (Existing(platform_config_path), Existing(home_config_path)) => Err(
-                ConfigEnvError::AmbiguousSource(platform_config_path, home_config_path),
-            ),
-            (Existing(path), _) | (_, Existing(path)) => Ok(Existing(path)),
-            (New(path), _) | (_, New(path)) => Ok(New(path)),
-            (Unavailable, Unavailable) => Ok(Unavailable),
+        if matches!(home_config_path, Existing(_))
+            || (matches!(home_config_path, New(_)) && matches!(platform_config_path, Unavailable))
+        {
+            paths.push(home_config_path);
         }
+        // This should be the default config created if there's
+        // no user config and `jj config edit` is executed.
+        if let Existing(_) | New(_) = platform_config_path {
+            paths.push(platform_config_path);
+        }
+        if let Existing(_) = platform_config_dir {
+            paths.push(platform_config_dir);
+        }
+        if paths.is_empty() {
+            paths.push(Unavailable);
+        }
+        Ok(paths)
```

Comment:
An empty vec should mean there are no config paths available, which overlaps with `ConfigPath::Unavailable`.

I think I have unsent patch that replaces `ConfigPath::Unavailable` with `None`. It might be better to upstream this patch first.

Reply:
Yeah, I had the same thought but decided not to remove Unavailable at the time.

However, I've gone ahead and removed it in this PR (so #6012) should no longer be needed---given the number of changes in this file I figured it was easier to implement it here than try to rebase and deal with the conflicts.

Reply:
We prefer small incremental changes, so it would be nice if you can rebase this on top (or split out a similar change.) I don't think this patch is big, though.
I think you can just rebase and restore the contents.

Reply:
So just to confirm, you want me to rebase my commit onto #6012?